cjs/src/api/services/demo.service.js

```diff
@@ -25,13 +25,18 @@ function deleteDemoItemService (paramsId) {
   return query;
 };
 
-function updateOneDemoItemPropertyValueService (paramsId, requestBody) {
-  const updateOps = {};
-  for (const ops of requestBody) {
-    updateOps[ops.propName] = ops.value;
-  }
-  const query = Demo.updateOne({ _id: paramsId }, { $set: updateOps }).exec();
-  return query;
+async function updateOneDemoItemPropertyValueService (paramsId, requestBody) {
+  const queryDoc = await Demo.findById(paramsId).select('_id name age').exec();
+
+  Object.keys(requestBody).forEach(key => {
+    if(key in queryDoc)
+      queryDoc[key] = requestBody[key];
+    else 
+      throw new Error(`property '${key}' does not exist in the mongoose model`)
+  });
+
+  const save = await queryDoc.save();
```

Comment:
**Review comment 4:** Instead of async - await for `cjs` template, use .then()

Reply:
Wow. I'll look into it. I actually had a lot of issues using .then(). I'll give it a try again.

Reply:
What specific issues did you have @samuko-things?

Reply:
> What specific issues did you have @samuko-things?

I'll send screenshots or short videos on the discord channel